tests/splitid.py

```diff
@@ -0,0 +1,196 @@
+def wait_for_identity(primary, retry_timeout):
+    have_identity = False
+    public_key = ""
+    with primary.client(timeout=60) as c:
+        timeout_before = time()
+        while time() < timeout_before + retry_timeout:
+            res = c.post("/node/splitid/get-identity", timeout=60)
+            if (
+                res.status_code == http.HTTPStatus.NOT_FOUND.value
+                or res.status_code == http.HTTPStatus.SERVICE_UNAVAILABLE.value
+            ):
+                sleep(1)
+            elif res.status_code == http.HTTPStatus.OK:
+                have_identity = True
+                public_key = res.body.json()["pem"]
+                break
+            else:
+                break
+    if not have_identity:
+        raise RuntimeError("split identity not established in time")
+    return public_key
+
+
+def run(args):
+    txs = app.LoggingTxs("user0")
+    with infra.network.network(
+        args.nodes,
+        args.binary_dir,
+        args.debug_nodes,
+        args.perf_nodes,
+        pdb=args.pdb,
+        txs=txs,
+    ) as network:
+        network.start_and_join(args)
+        primary, _ = network.find_primary()
+
+        retry_timeout = 120
+        defensive = False
+
+        msg = "themessage".encode("ascii")
+
+        # Wait until the split identity is available
+        public_key = wait_for_identity(primary, retry_timeout)
+
+        bench(
+            primary,
+            msg,
+            num_signatures=10,
+            defensive=defensive,
+            retry_timeout=retry_timeout,
+            public_key=public_key,
+        )
+
+        # trigger resharing by adding a node
+        new_node = network.create_node("local://localhost")
+        args.timeout = 60
+        network.join_node(new_node, args.package, args, from_snapshot=False)
+        network.trust_node(new_node, args)
+        with new_node.client() as c:
+            s = c.get("/node/state")
+            assert s.body.json()["node_id"] == new_node.node_id
+            assert (
+                s.body.json()["startup_seqno"] == 0
+            ), "Node started without snapshot but reports startup seqno != 0"
+        assert new_node
+
+        with primary.client() as c:
+            timeout_before = time()
+            while time() < timeout_before + retry_timeout:
+                s = c.get("/node/consensus")
+                rj = s.body.json()
+                if len(rj["details"]["configs"]) > 1:
+                    LOG.info("reconfiguration still in progress")
+                    sleep(1.0)
+                else:
+                    cfg = rj["details"]["configs"][0]
+                    if new_node.node_id in rj["details"]["configs"][0]["nodes"]:
+                        LOG.info(f"active config: {cfg}")
+                        break
+                    else:
+                        LOG.info(f"configruation does not contain new node yet: {cfg}")
+                        sleep(1.0)
+
+        # Wait until resharing is done
```

Comment:
Putting a comment here for reference, though it may not change anything in this PR: The next line doesn't actually wait at all (the earlier loop believes resharing is done, then the next line will return instantly since _an_ identity already exists to be returned, and always will be returned). Instead of this endpoint, we'd really like to confirm this by examining the network cert available from existing endpoints. We should put a custom claim in the x509, at least saying "this is _a_ shared identity", and ideally "this is shared between _these holders_" somehow, so that this reshare-due-to-reconfiguration can be confirmed.